packages/web3-react/src/components/AlephiumConnect.tsx

```diff
@@ -148,18 +171,21 @@ export const AlephiumConnectProvider: React.FC<{
 
   useEffect(() => {
     const func = async () => {
+      if (connectionStatusRef.current !== 'connecting') return
+
       const onDisconnected = () => {
```

Comment:
Q: If a connector doesn't have `autoConnect` defined, e.g. `WalletConnect`, and the `signer` was `undefined` before. Is it possible that `connectionStatus` is stuck at `connecting`?

Reply:
No, if auto connect fails, it will call the `onDisconnected` callback, which set the status to `disconnected`.

Reply:
In the latest code here: https://github.com/alephium/alephium-web3/pull/257/files#diff-d9969eee0099bfba3bd0dd6f677ddfc279b75331f2cf79de5f1545bac4315cb9R150, it sets the `connectionStatus` to `disconnected` if the `newSignerProvider` is undefined.